hail/src/main/scala/is/hail/expr/ir/Compile.scala

```diff
@@ -43,8 +43,11 @@ object Compile {
       .zipWithIndex
       .foldLeft(Env.empty[IR]) { case (e, ((n, t, _), i)) => e.bind(n, In(i, t)) }))
     ir = LoweringPipeline.compileLowerer.apply(ctx, ir, optimize).asInstanceOf[IR]
+
     TypeCheck(ir, BindingEnv.empty)
-    InferPType(if (HasIRSharing(ir)) ir.deepCopy() else ir, Env(args.map { case (n, pt, _) => n -> pt}: _*))
+
+    ir = if(HasIRSharing(ir)) ir.deepCopy() else ir
```

Comment:
we weren't actually setting _pType2 on the ir we passed to Emit (were setting and throwing away that IR copy)

Reply:
don't change that here, that's out of scope. ptype2 is totally unused in emit, we're just running the inference to make sure we don't break it in any PR.